pkg/logging/syslog/create.go

```diff
@@ -0,0 +1,74 @@
+package syslog
+
+import (
+	"io"
+
+	"github.com/fastly/cli/pkg/common"
+	"github.com/fastly/cli/pkg/compute/manifest"
+	"github.com/fastly/cli/pkg/config"
+	"github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/cli/pkg/text"
+	"github.com/fastly/go-fastly/fastly"
+)
+
+// CreateCommand calls the Fastly API to create Amazon Syslog logging endpoints.
+type CreateCommand struct {
+	common.Base
+	manifest manifest.Data
+	Input    fastly.CreateSyslogInput
+
+	// We must store all of the boolean flags seperatly to the input structure
+	// so they can be casted to go-fastly's custom `Compatibool` type later.
+	useTLS bool
```

Comment:
It should be possible to achieve this using a Kingpin custom parser instead of having to cast the value in the exeec function logic. Have a look at this and let me know your thoughts: https://github.com/alecthomas/kingpin#custom-parsers

Reply:
Thanks for sharing! I did look into this and I personally prefer the unexported field. I don't have much experience with the alecthomas/kingpin package, but it looks like the custom parser would have to be for the value that you are parsing _into_, in this case, a `fastly.CompatiBool`. With that said, it might be useful to have in a common package in the CLI for continued use of parsing CLI flags into `fastly.Compatibool`s. However, I still think I prefer the unexported field for a few reasons:

1. Less code to manage.
2. Introducing additional exported types/functions to a `common` package (or having duplicate, unexported implementations).
3. If we were to format it as recommended, the flag parsing would look inconsistent. For example,

```golang
	c.CmdClause.Flag("port", "The port number").UintVar(&c.Input.Port)
	CompatiBool(c.CmdClause.Flag("use-tls", ...
	c.CmdClause.Flag("tls-ca-cert",...
```

Reply:
Makes sense, I'm happy to leave as-is for now and we can refactor into the `common` package later if we feel it makes sense. :+1: